AIO/src/org/aio/activities/quests/SheepShearer.java

```diff
@@ -0,0 +1,150 @@
+import org.osbot.rs07.api.model.NPC;
+import org.osbot.rs07.api.model.RS2Object;
+import org.osbot.rs07.api.ui.Tab;
+
+public class SheepShearer extends QuestActivity {
+
+    private MakeAllInterface makeAllInterface;
+
+    private static final Area FARMER = new Area(3188, 3275, 3190, 3270);
+    private static final Area SHEEP = new Area(3195, 3272, 3209, 3260);
+    private static final Area SPINNER = new Area(3209, 3215, 3211, 3212).setPlane(1);
+
+    private static final String[] FARMER_OPTIONS = {
+            "I'm looking for a quest.",
+            "Yes okay. I can do that.",
+            "Of course!",
+            "I'm something of an expert actually!"
+    };
+
+    private static final int INVENTORY_SLOTS_REQUIRED = 22;
+
+    private static final String[] ITEMS_NEEDED = {
+            "Ball of wool",
+            "Wool",
+            "Shears"
+    };
+
+    private final DepositAllBanking depositAllBanking = new DepositAllBanking(ITEMS_NEEDED);
+
+    public SheepShearer() {
+        super(Quest.SHHEP_SHEARER);
+    }
+
+    @Override
+    public void onStart() {
+        depositAllBanking.exchangeContext(getBot());
+
+        makeAllInterface = new MakeAllInterface("Ball of Wool");
+        makeAllInterface.exchangeContext(getBot());
+    }
+
+    @Override
+    public void runActivity() throws InterruptedException {
+        if (getInventory().getEmptySlotCount() < INVENTORY_SLOTS_REQUIRED - getInventory().getAmount(ITEMS_NEEDED)) {
+            depositAllBanking.run();
+        } else if (getTabs().getOpen() != Tab.INVENTORY) {
+            getTabs().open(Tab.INVENTORY);
+        } else {
+            switch (getProgress()) {
+                case 0:
+                    talkToFarmer();
+                    break;
+                case 1:
+                    if (hasRequiredItems()) {
+                        talkToFarmer();
+                    } else {
+                        getItemsNeeded();
+                    }
+                    break;
+                case 21:
+                    log("Quest is complete");
+                    isComplete = true;
+                    break;
+                default:
+                    log("Unknown progress config value: " + getProgress());
+                    setFailed();
+                    break;
+            }
+        }
+    }
+
+    private boolean hasRequiredItems() {
+        return getInventory().getAmount("Ball of wool") > 19;
+    }
+
+    private void getItemsNeeded() throws InterruptedException {
+        if (!getInventory().contains("Shears")) {
+            getGroundItem(FARMER, "Shears");
+        } else if (!getInventory().contains("Ball of wool") && getInventory().getAmount("Wool") < 20) {
+            getItemFromNPC(SHEEP, "Shear");
+        } else if (getInventory().getAmount("Ball of wool") < 20) {
+            useObject(SPINNER, "Spin");
+        }
+    }
+
+    private void getItemFromNPC(Area place, String interaction) throws InterruptedException {
```

Comment:
Perhaps you could make this function slightly more generic with an NPC name and pull it out into a helpers function for questing? Not really a requirement though, just an idea.